mutable_tree.go

```diff
@@ -973,10 +999,13 @@ func (tree *MutableTree) balance(node *Node) (newSelf *Node, err error) {
 func (tree *MutableTree) saveNewNodes(version int64) error {
 	nonce := int32(0)
 	newNodes := make([]*Node, 0)
-	var recursiveAssignKey func(*Node) (*NodeKey, error)
-	recursiveAssignKey = func(node *Node) (*NodeKey, error) {
+	var recursiveAssignKey func(*Node) ([]byte, error)
+	recursiveAssignKey = func(node *Node) ([]byte, error) {
 		if node.nodeKey != nil {
-			return node.nodeKey, nil
+			if node.nodeKey.nonce > 0 {
```

Comment:
`nonce` is a signed `int32` so I expected `if node.nodeKey.nonce != 0`. Raised https://github.com/cosmos/iavl/issues/763 to avoid similar issues.